### PR TITLE
Limits some monitors' modes due to hardware limitation.

### PR DESCRIPTION
**Pull request by yugang**
Limits some monitors' modes due to hardware limitation.

### CODE REVIEW

**Comment by yugang** [comment]
@js0701 @YuanjunHuang @Shao-Feng , pls check if it can be used to indentify view sonic display.

**Comment by yugang** [comment]
debug log just as  “Monitor name: TD2230 Series, manufacturer: VSC”

File: wsi/edid.h [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+#ifndef eds_edid_h
+#define eds_edid_h
+
+#include <assert.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#define ARRAY_SIZE(arr)                         (sizeof(arr) / sizeof(arr[0]))
+
+namespace hwcomposer {
```
Comment (js0701) [inline wsi/edid.h]: you changed this header file?

Comment (yugang) [reply wsi/edid.h]: Yes,removed some unused code and added one simple func

**Comment by yugang** [comment]
Added the original place and modification details of edid.h in comment area.

**Comment by yugang** [comment]
updated the commit:
1. removed the third party code.
2. following the EDID spec, updated the code and struct of display description data of EDID

**Comment by yugang** [comment]
pls help review and merge it, thank you

File: wsi/drm/drmdisplay.cpp [hunk 2]
```diff
@@ -154,6 +154,49 @@ void DrmDisplay::DrmConnectorGetDCIP3Support(
...
+    if (display_data[i].tag_number == 0xfc) {
+      display_name_.clear();
+      display_name_.assign((char *)display_data[i].desc_data, 13);
+    }
+  }
+
+  ITRACE("EDID Display Name: %s\n", display_name_.c_str());
+}
+
+bool DrmDisplay::CheckLimitedMonitor() {
```
Comment (YuanjunHuang) [inline wsi/drm/drmdisplay.cpp]: pls add comment for the aim of this piece of code  is that to w/a certain display which is not capable for the retrieved modes it declares.

**Comment by js0701** [comment]
Thanks @yugang

File: wsi/drm/drmdisplay.cpp (hunk 2 above)
Comment (js0701) [reply wsi/drm/drmdisplay.cpp]: And we still can have those fixed constant defined by MACRO, it is more meaningful, right

**Comment by js0701** [comment]
@yugang 
I see setMode is limited,
but how about getMode? We should avoid expose those invalid modes to AOSP

File: wsi/drm/drmdisplay.cpp (hunk 2 above)
Comment (YuanjunHuang) [reply wsi/drm/drmdisplay.cpp]: > And we still can have those fixed constant defined by MACRO, it is more meaningful, right

agree

**Comment by yugang** [comment]
@js0701 , here we only expose the one prefered mode to mode_ from modes supportted by i915 connector(as UpdateDisplayState is the only place to init the mode_, we just limited here), and all APIs of HWC will only base on "mode_" of the display, whatever getactiveconfig/setactiveconfig/GetDisplayConfigs/UpdateDisplayConfig/SetDrmModeInfo/current_mode_/.

**Comment by js0701** [comment]
OK

**Comment by yugang** [comment]
@YuanjunHuang @js0701:
1. add comments for the CheckLimitedMonitor 
2. define a string vector HWC_LIMITED_MONITOR_LIST for future appending, e.g. HWC_LIMITED_MONITOR_LIST{"TD2230 Series", "DELL U2414H"}

**Comment by YuanjunHuang** [comment]
LGTM.

**Comment by YuanjunHuang** [comment]
please follow https://github.com/intel/IA-Hardware-Composer/wiki/Conformance to fix the coding style conformance issue.

**Comment by yugang** [comment]
@YuanjunHuang seems it is travis build issue:

`
configure: error: Package requirements (libinput >= 0.8.0) were not met:

No package 'libinput' found
`

**Comment by yugang** [comment]
pls hold the merging as i found some diff between android and linux

**Comment by yugang** [comment]
updated done: resize the display name string instead of using '\0'

**Comment by yugang** [comment]
added issue tracking on

**Pull request merged**